web_tree_many2one_clickable/static/src/components/many2one_button/many2one_button.esm.js

```diff
@@ -0,0 +1,34 @@
+/** @odoo-module */
+/* Copyright 2013 Therp BV (<http://therp.nl>).
+ * Copyright 2015 Pedro M. Baeza <pedro.baeza@serviciosbaeza.com>
+ * Copyright 2016 Antonio Espinosa <antonio.espinosa@tecnativa.com>
+ * Copyright 2017 Sodexis <dev@sodexis.com>
+ * Copyright 2018 Camptocamp SA
+ * Copyright 2019 Alexandre Díaz <alexandre.diaz@tecnativa.com>
+ * Copyright 2024 Versada (https://versada.eu)
+ * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). */
+
+import {ListRenderer} from "@web/views/list/list_renderer";
+import {Component} from "@odoo/owl";
+import {useService} from "@web/core/utils/hooks";
+
+export class TreeMany2oneClickableButton extends Component {
+    setup() {
+        this.actionService = useService("action");
+    }
+
+    async onClick(ev) {
+        ev.stopPropagation();
+        return this.actionService.doAction({
+            type: "ir.actions.act_window",
+            res_model: this.props.field.relation,
+            res_id: this.props.value[0],
+            views: [[false, "form"]],
+            target: "target",
+            additionalContext: this.props.context || {},
+        });
+    }
+}
+TreeMany2oneClickableButton.template = "web_tree_many2one_clickable.Button";
+
+Object.assign(ListRenderer.components, {TreeMany2oneClickableButton});
```

Comment:
Was this done originally? Shouldn't we use "patch" to modify existing component?

Reply:
It is a part of the code included in migration to v16. I will investigate if we can use `patch`

Reply:
@oerp-odoo I couldn't find any examples of extending components of existing class via `patch`. But `Object.assign` is quite commonly used [[example](https://github.com/versada/odoo/blob/9dba4e0d77187f0158243d314057f7aa3a19b543/addons/mail/static/src/discuss/call/common/chat_window_patch.js#L6)]

Reply:
suggest not to change if it works,  but opportunity to  try understand js specifics:

docs have example for [static properties](https://www.odoo.com/documentation/17.0/developer/reference/frontend/patching_code.html?highlight=static%20properties#patching-a-javascript-class), 

But [js static properties](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/static) are not propagated to instances (like in python)..

Still class methods can be patched via `prototype` (in same docs), maybe `components` property could also...